### PR TITLE
Track2 Migration: Adding headers and query params to be logged

### PR DESCRIPTION
**Pull request by souravgupta-msft**

### CODE REVIEW

File: component/azstorage/azstorage_constants.go [hunk 1]
```diff
@@ -60,3 +60,14 @@ const (
 	size        = "Size"
 	target      = "Target"
 )
+
+// headers which should be logged and not redacted
+var allowedHeaders []string = []string{
+	"x-ms-version", "x-ms-date", "x-ms-range", "x-ms-delete-snapshots", "x-ms-delete-type-permanent", "x-ms-blob-content-type",
```
Comment (vibhansa-msft) [inline component/azstorage/azstorage_constants.go:66]: request ids ? they will be important for backend tracing.

File: component/azstorage/utils.go [hunk 2]
```diff
@@ -131,8 +131,8 @@ func getSDKLogOptions() policy.LogOptions {
 	} else {
 		// TODO:: track2 : check which headers and query params should not be redacted
 		return policy.LogOptions{
-			AllowedHeaders:     []string{"x-ms-version"},
-			AllowedQueryParams: []string{},
+			AllowedHeaders:     allowedHeaders,
```
Comment (vibhansa-msft) [inline component/azstorage/utils.go:134]: by default does it log something or everything is redacted ?

File: component/azstorage/azstorage_constants.go (hunk 1 above)
Comment (souravgupta-msft) [reply component/azstorage/azstorage_constants.go:66]: Request ids are logged by default.

File: component/azstorage/utils.go (hunk 2 above)
Comment (souravgupta-msft) [reply component/azstorage/utils.go:134]: By default, these headers are logged, https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/runtime/policy_logging.go#L38-L64

And these query params are logged, https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/runtime/policy_logging.go#L81-L83

Comment (souravgupta-msft) [reply component/azstorage/utils.go:134]: The list we provided here are the additional headers and query params that we want to be logged and not redacted.

**Review by vibhansa-msft — APPROVED** [review]

**Review by ashruti-msft — APPROVED** [review]

**Pull request merged**